vdslib/src/vespa/vdslib/distribution/distribution.cpp

```diff
@@ -378,46 +375,46 @@ Distribution::getIdealNodes(const NodeType& nodeType, const ClusterState& cluste
             ss << "There is no legal distributor target in state with version " << clusterState.getVersion();
             throw NoDistributorsAvailableException(ss.str(), VESPA_STRLOC);
         }
-        _groupDistribution.push_back(ResultGroup(*group, 1));
+        _groupDistribution.emplace_back(*group, 1);
     }
     RandomGen random(seed);
     uint32_t randomIndex = 0;
     std::vector<ScoredNode> tmpResults;
-    for (uint32_t i=0, n=_groupDistribution.size(); i<n; ++i) {
-        uint16_t groupRedundancy(_groupDistribution[i]._redundancy);
-        const std::vector<uint16_t>& nodes(_groupDistribution[i]._group->getNodes());
+    for (const auto & group : _groupDistribution) {
+        uint16_t groupRedundancy(group._redundancy);
+        const std::vector<uint16_t>& nodes(group._group->getNodes());
         // Create temporary place to hold results.
         // Stuff in redundancy fake entries to
         // avoid needing to check size during iteration.
         tmpResults.reserve(groupRedundancy);
         tmpResults.clear();
         tmpResults.resize(groupRedundancy);
-        for (uint32_t j=0; j < nodes.size(); ++j) {
+        for (unsigned short node : nodes) {
```

Comment:
Please use explicit `uint16_t` for types referencing distribution keys

Reply:
Fixed